odoo/addons/base/tests/test_ormcache.py

```diff
@@ -37,3 +37,63 @@ def test_ormcache(self):
         self.assertEqual(counter.hit, hit + 2)
         self.assertEqual(counter.miss, miss + 1)
         self.assertIn(key, cache)
+
+    def test_invalidation(self):
+        self.assertEqual(self.env.registry.cache_invalidated, set())
+        self.env.registry.clear_cache()
+        self.env.registry.clear_cache('templates')
+        self.assertEqual(self.env.registry.cache_invalidated, {'default', 'templates'})
+        self.env.registry.reset_changes()
+        self.assertEqual(self.env.registry.cache_invalidated, set())
+        self.env.registry.clear_cache('assets')
+        self.assertEqual(self.env.registry.cache_invalidated, {'assets'})
+        self.env.registry.reset_changes()
+        self.assertEqual(self.env.registry.cache_invalidated, set())
+
+    def test_invalidation_thread_local(self):
+        # this test ensures that the registry.cache_invalidated set is thread local
+
+        caches = ['default', 'templates', 'assets']
+        nb_treads = len(caches)
+
+        # use barriers to ensure threads synchronization
+        sync_clear_cache = Barrier(nb_treads, timeout=5)
+        sync_assert_equal = Barrier(nb_treads, timeout=5)
+        sync_reset = Barrier(nb_treads, timeout=5)
+
+        opperations = []
```

Comment:
One "p" is enough :wink: 
```suggestion
        operations = []
```